src/init.cpp

```diff
@@ -878,6 +879,7 @@ void InitLogging(const ArgsManager& args)
 #ifdef HAVE_THREAD_LOCAL
     LogInstance().m_log_threadnames = args.GetBoolArg("-logthreadnames", DEFAULT_LOGTHREADNAMES);
 #endif
+    LogInstance().m_log_functionnames = gArgs.GetBoolArg("-logfunctionnames", DEFAULT_LOGFUNCTIONNAMES);
```

Comment:
```suggestion
    LogInstance().m_log_functionnames = args.GetBoolArg("-logfunctionnames", DEFAULT_LOGFUNCTIONNAMES);
```

nit: Would be nice to not introduce more gArgs

Reply:
Good point. Fixed!